server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Capability.scala

```diff
@@ -32,12 +34,20 @@ object CapabilityIdentifier {
   val JMAP_MAIL = CapabilityIdentifier(new URI("urn:ietf:params:jmap:mail"))
 }
 
-sealed trait Capability
+sealed trait CapabilityProperties
+
+sealed trait Capability {
+  def identifier(): CapabilityIdentifier
+  def properties(): CapabilityProperties
+}
 
-final case class CoreCapability(maxSizeUpload: UnsignedInt, maxConcurrentUpload: UnsignedInt,
-                                maxSizeRequest: UnsignedInt, maxConcurrentRequests: UnsignedInt,
-                                maxCallsInRequest: UnsignedInt, maxObjectsInGet: UnsignedInt,
-                                maxObjectsInSet: UnsignedInt, collationAlgorithms: List[String]) extends Capability {
+final case class CoreCapability(identifier: CapabilityIdentifier = JMAP_CORE,
```

Comment:
seems strange to have parameters with default value before parameter without it.
In earlier version of scala the compiler didn't accept that.